ctx/ctx.go

```diff
@@ -133,3 +136,29 @@ func GetDefinition(r *http.Request) *apidef.APIDefinition {
 	}
 	return nil
 }
+
+func SetOASDefinition(r *http.Request, s *oas.OAS) {
```

Comment:
are we implementing `SetOASDefinition` at this point?

Reply:
we shouldn't (on ticket)

Reply:
Also if we were to implement this, do we need to do the Fill and extractTo?

Reply:
I didn't mean to implement the mentioned function in the refinement, I wanted to implement an internal setter, so I will move it to the middleware file and start with small case.